webviz_subsurface/_utils/vector_calculator.py

```diff
@@ -0,0 +1,325 @@
+
+    # Validate expression string
+    if not parsed_expression["isValid"]:
+        parse_message = parsed_expression["message"]
+        message = (
+            f'Invalid mathematical expression {expr} in predefined expression "{name}".'
+            f"{parse_message}."
+        )
+        return False, message
+
+    # Match variables in expression string and variable names in map
+    expression_variables = parsed_expression["variables"]
+    map_variables = [elm["variableName"] for elm in expression["variableVectorMap"]]
+    if set(expression_variables) != set(map_variables):
+        message = (
+            f"Variables {map_variables} in variableVectorMap is inconsistent with variables "
+            f'{expression_variables} in equation "{expr}" for predefined expression "{name}"'
+        )
+        return False, message
+
+    # Validate vector names
+    variable_vector_dict = VectorCalculator.variable_vector_dict(
+        expression["variableVectorMap"]
+    )
+    invalid_vectors: List[str] = []
+    for vector_name in variable_vector_dict.values():
+        if not is_vector_name_existing(vector_name, vector_data):
+            invalid_vectors.append(vector_name)
+    if len(invalid_vectors) > 1:
+        message = (
+            f'Vector names {invalid_vectors} in predefined expression "{name}" are not'
+            f" represented in vector data"
+        )
+        return False, message
+    if len(invalid_vectors) > 0:
+        message = (
+            f'Vector name {invalid_vectors} in predefined expression "{name}" is not'
+            f" represented in vector data"
+        )
+        return False, message
+
+    return True, ""
+
+
+def variable_vector_map_from_dict(
+    variable_vector_dict: Dict[str, str]
+) -> List[VariableVectorMapInfo]:
+    variable_vector_map: List[VariableVectorMapInfo] = []
+    for variable in variable_vector_dict:
+        variable_vector_map.append(
+            {
+                "variableName": variable,
+                "vectorName": [variable_vector_dict[variable]],
+            }
+        )
+    return variable_vector_map
+
+
+def expressions_from_config(
+    predefined_expressions: Optional[Path],
+) -> List[ExpressionInfo]:
+    output: List[ExpressionInfo] = []
+
+    if predefined_expressions is None:
+        return output
+
+    expressions: Dict[str, ConfigExpressionData] = yaml.safe_load(
+        predefined_expressions.read_text()
+    )
+
+    for expression in expressions:
+        expression_dict = {
+            "name": expression,
+            "expression": expressions[expression]["expression"],
+            "id": f"{uuid4()}",
+            "variableVectorMap": variable_vector_map_from_dict(
+                expressions[expression]["variableVectorMap"]
+            ),
+            "isValid": False,  # Set False and validate in seperate operation
+            "isDeletable": False,
+        }
+        if "description" in expressions[expression].keys():
```

Comment:
```suggestion
        if "description" in expressions[expression]:
```
...or maybe my suggestion does not work? (I'm a bit surprised `pylint`didn't suggest this).